src/Knp/Rad/ResourceResolver/Caster/ArrayCaster.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Knp\Rad\ResourceResolver\Caster;
+
+use Knp\Rad\ResourceResolver\Caster;
+
+final class ArrayCaster implements Caster
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    /**
+     * @param Container $container
+     */
+    public function __construct(Container $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function cast($array)
+    {
+        foreach ($array as $key => $value) {
+            $array[$key] = $this->container->cast($value);
+        }
+
+        return $array;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($array)
+    {
+        return true === is_array($array);
```

Comment:
Are you kidding me ?????

![r111q](https://cloud.githubusercontent.com/assets/1766827/9851578/d08d613c-5b12-11e5-9654-21a01aa3795b.jpg)